internal/runners/activate/activate.go

```diff
@@ -113,6 +120,37 @@ func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc)
 	return activatorLoop(r.out, r.subshell, projectPath, activate)
 }
 
+func getActualNamespace(replace bool, prj *project.Project, names *project.Namespaced) (*project.Namespaced, error) {
+	if replace {
+		if names.CommitID == nil || *names.CommitID == "" {
+			latestID, fail := model.LatestCommitID(names.Owner, names.Project)
+			if fail != nil {
+				return nil, locale.WrapInputError(fail.ToError(), "err_set_namespace_retrieve_commit", "Could not retrieve the latest commit for the specified project {{.V0}}.", names.String())
+			}
+			names.CommitID = latestID
+		}
+		err := prj.Source().SetNamespace(names.String())
+		if err != nil {
+			return nil, locale.WrapError(err, "err_activate_replace_write_namespace", "Failed to write new namespace to activestate.yaml.")
+		}
+		fail := prj.Source().SetCommit(names.CommitID.String())
+		if fail != nil {
+			return nil, locale.WrapError(fail.ToError(), "err_activate_replace_write_commit", "Failed to write commitID to activestate.yaml.")
+		}
+		return names, nil
+	}
+
+	projectPath := filepath.Dir(prj.Source().Path())
+
+	var fail *failures.Failure
+	nspace, fail := project.ParseNamespaceOrConfigfile(names.String(), filepath.Join(projectPath, constants.ConfigFileName))
+	if fail != nil {
+		nspace = &project.Namespaced{}
+		logging.Debug("error resolving namespace: %v", fail.ToError())
+	}
+	return nspace, nil
+}
```

Comment:
This is doing too much. It's doing both replace AND figuring out the namespace. The namespace stuff can stay where it was for all I care. It was the replace stuff that should have its own function.

Keep it unixy! ;)